src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,105 @@
+package duke;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+public class Storage {
+    private final String filePath;
+    private final File hardDiskFile;
+
+    /**
+     * Constructor for Storage.
+     *
+     * @param filePath Path of file that will be used as storage.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.hardDiskFile = new File(filePath);
+    }
+
+    private void createFile() {
+        try {
+            File parentDir = hardDiskFile.getParentFile();
+            if (!parentDir.exists()) {
+                parentDir.mkdirs();
+            }
+            hardDiskFile.createNewFile();
+        } catch (IOException e) {
+            System.out.println("Error when creating new file: " + e);
+        }
+    }
+
+    /**
+     * Returns an Arraylist of all tasks in the storage.
+     *
+     * @return Arraylist of tasks in the storage.
+     */
+    public ArrayList<Task> getTasks() {
+        createFile();
+
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                String[] parts = line.split("\\|");
+
+                if (parts.length < 3) {
+                    continue;
+                }
+
+                String identifier = parts[0];
+                boolean done = parts[1] == "1";
```

Comment:
I feel that this boolean should be named "isDone" to reflect that it is a boolean.